### PR TITLE
Update vm-25.kql

### PR DESCRIPTION
**Pull request by davenewman777**
# Overview/Summary

New APRL query for WM-25

## Related Issues/Work Items

https://dev.azure.com/CSUSolEng/Well-Architected%20Framework/_workitems/edit/34259

[AB#34259](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/34259)

## This PR fixes/adds/changes/removes

Creates a new query where one did not exist.

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by davenewman777** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: docs/content/services/compute/virtual-machines/_index.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ The presented resiliency recommendations in this guidance include Virtual Machin
 | [VM-22 - Use maintenance configurations for the Virtual Machine](#vm-22---use-maintenance-configurations-for-the-vms) | Governance | High | Preview | Yes |
 | [VM-23 - Avoid using A or B-Series VM Sku for production VMs that need the full performance of the CPU continuously](#vm-23---avoid-using-a-or-b-series-vm-sku-for-production-vms-that-need-the-full-performance-of-the-cpu-continuously) | System Efficiency | High | Preview | Yes |
 | [VM-24 - Mission Critical Workloads should be using Premium or Ultra Disks](#vm-24---mission-critical-workloads-should-be-using-premium-or-ultra-disks) | System Efficiency | High | Preview | Yes |
-| [VM-25 - Do not create more than 2500 Citrix VDA servers per subscription](#vm-25---do-not-create-more-than-2500-citrix-vda-servers-per-subscription) | Application Resiliency | High | Preview | Yes |
+| [VM-25 - Do not create more than 2500 Citrix VDA servers per subscription](#vm-25---do-not-create-more-than-2000-citrix-vda-servers-per-subscription) | Application Resiliency | High | Preview | Yes |
```
Comment (ejhenry) [inline docs/content/services/compute/virtual-machines/_index.md:41]:
```suggestion
| [VM-25 - Do not create more than 2000 Citrix VDA servers per subscription](#vm-25---do-not-create-more-than-2000-citrix-vda-servers-per-subscription) | Application Resiliency | High | Preview | Yes |
```

**Review by ejhenry — COMMENTED** [review]
Thanks @davenewman777! Can you update the query output to align with the standard in the contribution guide here: https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing/#azure-resource-graph-arg-queries

File: docs/content/services/compute/virtual-machines/code/vm-25/vm-25.kql [hunk 2]
```diff
@@ -1 +1,9 @@
-// under-development
+// Azure Resource Graph Query
+// Count VM instances with a tag that contains "Citrix VDA" and create output if that count is >2000 for each subscription.
+// Because this is a count of resources, ResourceID is not publishable
+resources
+| where type == 'microsoft.compute/virtualmachines'
+| where tags contains 'Citrix VDA'
+| summarize VMs=count() by subscriptionId
+| where VMs > 2000
```
Comment (tksh164) [inline docs/content/services/compute/virtual-machines/code/vm-25/vm-25.kql:8]: Is `2000` correct? The linked document in resource section says `2500`.

https://azure.github.io/Azure-Proactive-Resiliency-Library/services/compute/virtual-machines/#vm-25---do-not-create-more-than-2500-citrix-vda-servers-per-subscription

image

image

**Review by tksh164 — COMMENTED** [review]
Hi @davenewman777, the limitation number in the query looks different from the documented limit. Which is the correct value?

**Comment by prasad3017** [comment]
Hello @tksh164  it is 2500 from Citrix documentation.  CC: @davenewman777

**Review by oZakari — CHANGES_REQUESTED** [review]
Hey @davenewman777, as we discussed last week, could we please create a new service recommendation page called Subscription and move this to there since it is not specific to the configuration of a vm. You can put it underneath the Management service category.

**Comment by ejhenry** [comment]
To @tksh164's point, I think it makes sense to call out in the recommendation that we are saying customers should not exceed 80% of the Citrix recommended limit.

File: docs/content/services/compute/virtual-machines/_index.md (hunk 1 above)
Comment (davenewman777) [reply docs/content/services/compute/virtual-machines/_index.md:41]: adjustments made as per our conversation

File: docs/content/services/compute/virtual-machines/code/vm-25/vm-25.kql (hunk 2 above)
Comment (tksh164) [reply docs/content/services/compute/virtual-machines/code/vm-25/vm-25.kql:8]: I could understand why the query used 2000. I think more better to add that reason (80% of the Citrix limit) into the recommendation's guidance.

**Comment by davenewman777** [comment]
not changing VM-25 - creating new instead

File: docs/content/services/management/subscription/_index.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+**Guidance**
+
+A Citrix Managed Azure subscription supports the number of machines indicated in Limits. (In this context, machines refers to VMs that have a Citrix VDA installed. These machines deliver apps and desktops to users. It does not include other machines in a resource location, such as Cloud Connectors.)
+
+If your Citrix Managed Azure subscription is likely to reach its limit soon, and you have enough Citrix licenses, you can request another Citrix Managed Azure subscription. The dashboard contains a notification when you’re close to the limit.
+
+You can’t create a catalog (or add machines to a catalog) if the total number of machines for all catalogs that use that Citrix Managed Azure subscription would exceed the value indicated in Limits.
+
+This recommendation checks for 80% of the Citrix limit so that attention can be paid to this before hitting the published Citrix limit of 2500.
```
Comment (oZakari) [inline docs/content/services/management/subscription/_index.md:43]:
```suggestion
This recommendation ensures that when 80% of the Citrix limit is reached, it prompts attention to be given before reaching the maximum Citrix limit of 2500.
```

**Review by oZakari — APPROVED** [review]

**Review by oZakari — APPROVED** [review]

**Review by ejhenry — APPROVED** [review]
LGTM 👍

**Pull request merged**